app/ui-react/packages/auto-form/src/widgets/FormDurationComponent.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  ControlLabel,
+  DropdownButton,
+  Form,
+  FormGroup,
+  HelpBlock,
+  MenuItem,
+} from 'patternfly-react';
+import * as React from 'react';
+import { IFormControl } from '../models';
+import { getValidationState } from './helpers';
+
+interface IDuration {
+  label: string;
+  value: number;
+}
+
+const durations = [
+  {
+    label: 'Milliseconds',
+    value: 1,
+  },
+  {
+    label: 'Seconds',
+    value: 1000,
+  },
+  {
+    label: 'Minutes',
+    value: 60000,
+  },
+  {
+    label: 'Hours',
+    value: 3600000,
+  },
+  {
+    label: 'Days',
+    value: 86400000,
+  },
+] as IDuration[];
+
+export interface IFormDurationComponentState {
+  duration: IDuration;
+  value: number;
+}
+
+export class FormDurationComponent extends React.Component<
+  IFormControl,
+  IFormDurationComponentState
+> {
+  private inputField: HTMLInputElement = undefined as any;
+  constructor(props: IFormControl) {
+    super(props);
+    const index =
+      durations.findIndex(d => !(this.props.field.value / d.value >= 1)) - 1;
+    const duration = durations[index] || durations[0];
+    const value = this.calculateDuration(duration, this.props.field.value);
```

Comment:
I think this value copy leads to stale data when the component receives a new value since there is no `onComponentDidUpdate` handler. 

I don't think you want to have a copy of the value in the state at all. The calculation could be done in the render method, so to always use the value passed in the props. Duration is ok to be kept in the state since it only changes the way the value is displayed to the user.